README.md

```diff
@@ -82,9 +82,39 @@ System landscape view where Valkyrie fits in:
 Components view of Valkyrie, how the internals connect to each other.
 ![components view](/structurizr/structurizr-1-Valkyrie-Components.png)
 
+### Codebase structure
+Going through the code there are two folders that are more important than others, `<providers>` and `<pam>`. In these you will find the relevant components that makes Valkyrie modular. 
+In `<providers>` you will find each available game provider module that is integrated with Valkyrie. More in detail how to create a new provider module and the code structure of them can be found [here](./example/README.md). 
+When starting a Valkyrie instance you provide it with a config yaml file containing what provider modules you want to have enabled. Read more about the configuration file [here](https://valkyrie.bet/docs/get-started/configuration).
+In the root `<provider>`-folder is the implementations of `provider/docs/operator_api.yml`. It is the specification the operator can use to send request to Valkyrie that will be forwarded toward the provider. You can read more [here](https://valkyrie.bet/docs/operator/valkyrie-operator-api).
+
+`<pam>` contains implemented PAMs that are available. Each provider is using `pam.PamClient` found in `/pam/client.go` to communicate with the PAM. The available PAMs should implement this interface. Similar to the providers, what PAM is used is based on the configuration file provided at runtime. There is an oapi specification that is used by `<genericpam>` called `pam_api.yml` that can be read [here](https://valkyrie.bet/docs/wallet/valkyrie-pam/valkyrie-pam-api). 
+Fulfilling this specification will enable you to use `<genericpam>`. In the diagram above, `<genericpam>` and `<vplugin>` would be the "PAM Client". In the case of `<vplugin>` is when "PAM Plugin" would also be used. `<vplugin>` is utilizing hashicorp go-plugin to keep the pam implementation in a separate process from Valkyrie. More on this can be read [here](https://valkyrie.bet/docs/wallet/vplugin/vplugin-introduction).
+``` text
++ configs/          # The parsing of configuration yaml
++ example/          # An example provider module information about the different components needed to make your own provider module
```

Comment:
Just wondering, did you consider placing this under `provider/`?

Reply:
I think I wanted to keep `example` at the root level for discoverability when viewing the repo on github. With that said I think it could contain more examples than just an example provider. I'll clarify that a bit